scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -682,25 +682,23 @@ def _response_constraints_validation(self, response, result):
         check_type = PostProcessCheckType.CONSTRAINT_VALIDATION
         error_success = 'Constraints check passed'
         error_failure = 'Constraints check failed'
-        error_name_does_not_exist = 'The test expects a value named "{name}" but it does not exists in the response."'
+        error_name_does_not_exist = 'Constraint check expected a value named "{name}" but it does not exist in the response."'
 
         for value in self.response['values']:
             if 'constraints' not in value:
                 continue
 
-            expected_name = 'value'
             received_value = response.get('value')
             if not self.is_attribute:
                 expected_name = value.get('name')
                 if received_value is None or expected_name not in received_value:
-                    result.error(check_type, error_name_does_not_exist.format(
-                        name=expected_name))
-                    continue
-
-                received_value = received_value.get(
-                    expected_name) if received_value else None
+                    received_value = None
+                else:
+                    received_value = received_value.get(
+                        expected_name) if received_value else None
```

Comment:
When is `received_value` falsy here (such that we would assign literal None to `received_value`)?

Reply:
TBH I don't remember exactly right now. IIRC it could be an empty dictionary

Reply:
Then `expected_name not in received_value` would be false and we would not reach this branch, right?

Reply:
TBH I don't remember right now, maybe this is an optimization we could make, but for the time being I don't see the harm keeping this identical to what was here before

Reply:
That's fine; I just wanted to understand what the actual data flowing through here is...